### PR TITLE
feat(notebook): Support Launching Notebooks with CRD Directly

### PR DESCRIPTION
**Pull request by Mirrored90**

### CODE REVIEW

File: pkg/notebook/generate.go [hunk 1]
```diff
@@ -37,27 +38,66 @@ const (
...
+		log: l,
 	}
 
 	return g, nil
 }
 
 func (g generator) DesiredDeploymentWithoutOwner() *appsv1.Deployment {
+	if g.nb.Spec.Template == nil && g.nb.Spec.Gateway == nil {
+		// TODO (Qingyu Wu): error handle
+		panic("You must apply a gateway or template")
```
Comment (gaocegege) [inline pkg/notebook/generate.go:62]: We should not panic here. How about issuing an error?

**Review by gaocegege — COMMENTED** [review]
Please add some test cases.

Thanks for your contribution! :tada: :+1:

File: pkg/notebook/reconcile.go [hunk 2]
```diff
@@ -65,7 +67,11 @@ func (r Reconciler) Reconcile() error {
 }
 
 func (r Reconciler) reconcileDeployment() error {
-	desired := r.gen.DesiredDeploymentWithoutOwner()
+	desired, err_ := r.gen.DesiredDeploymentWithoutOwner()
```
Comment (gaocegege) [inline pkg/notebook/reconcile.go:70]: Please use err instead of err_

File: pkg/notebook/reconcile.go [hunk 3]
```diff
@@ -65,7 +67,11 @@ func (r Reconciler) Reconcile() error {
...
+	if err_ != nil {
+		r.log.Error(err_, "Failed to get desired deployment")
```
Comment (gaocegege) [inline pkg/notebook/reconcile.go:72]: No need to log it since we will print the error outside.

Comment (Mirrored90) [reply pkg/notebook/reconcile.go]: Fixed

File: pkg/notebook/reconcile.go (hunk 2 above)
Comment (Mirrored90) [reply pkg/notebook/reconcile.go]: Fixed

**Review by gaocegege — APPROVED** [review]

**Pull request merged**